### PR TITLE
fix(modal): resolve findDOMNode deprecation warning in React 17 projects

### LINKED ISSUE #4585: Dialog component initialization issue
### Current behaviour

In some cases Dialog component is not displayed.

#### React 17
In react 17, navigating directly to `/popup` the dialog is not displayed when button is clicked.
When navigate to the page using a navlink, the dialog is displayed when button is clicked.

https://codesandbox.io/s/vigorous-shockley-11bmh
https://11bmh.csb.app/popup

#### React 16
This works as expected with react 16.
https://codesandbox.io/s/suspicious-phoebe-e9zfz
https://e9zfz.csb.app/popup

### Expected behaviour

Dialog is displayed in any cases

### CodeSandbox or Storybook URL

https://codesandbox.io/s/vigorous-shockley-11bmh

### JIRA Ticket (Sage Only)

_No response_

### Suggested Solution

_No response_

### Carbon Version

100.1.1

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

Windows

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour
Pass down a ref to an animated container in the nodeRef prop of the Transition.

### Current behaviour

Deprecation warning is triggered by react-transition-group internally, when it is trying to find the container to be animated.

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

fixes #4585

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

**Review by mkrds — APPROVED** [review]

**Review by mkrds — DISMISSED** [review]

**Review by jamime — DISMISSED** [review]

**Review by mkrds — DISMISSED** [review]

**Review by mkrds — APPROVED** [review]

File: src/components/portal/portal.js [hunk 1]
```diff
@@ -53,15 +57,15 @@ const Portal = ({ children, className, id, onReposition }) => {
...
+      let mainNode = document.body;
       const rootDiv = document.getElementById("root");
 
-      // Storybook adds the hidden attribute to the root element when on the "Docs" pages
-      // Without this check the portal would also be hidden
-      if (rootDiv && !rootDiv.hasAttribute("hidden")) {
-        rootDiv.appendChild(node);
-      } else {
-        document.body.appendChild(node);
+      if (rootDiv && isChromatic()) {
```
Comment (jamime) [inline src/components/portal/portal.js:63]: I don't want us to add the `chromatic` package as a dependency. Is there another way to achieve this?

I think at a minimum we should implement a context provider to abstract this away from our consumers, we can then wrap our stories in the context provider that provides the chromatic specific logic.

File: cypress/integration/common/menu.feature [hunk 2]
```diff
@@ -119,26 +119,30 @@ Feature: Menu component
     When I click closeIcon
     Then Menu is in fullscreen mode is not visible
 
-  @positive
+  @ignore
```
Comment (jamime) [inline cypress/integration/common/menu.feature:122]: Please can you add the JIRA ticket reference for the new task to these comments.

**Review by jamime — CHANGES_REQUESTED** [review]

File: src/components/portal/portal.js (hunk 1 above)
Comment (grabkowski) [reply src/components/portal/portal.js]: good point, I have wrapped it in a context, and the value is set in storybook now

**Review by mkrds — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**